index.d.ts

```diff
@@ -1,5 +1,14 @@
 type Awaited<ValueType> = ValueType extends undefined ? ValueType : ValueType extends PromiseLike<infer ResolveValueType> ? ResolveValueType : ValueType;
 
+// https://github.com/microsoft/TypeScript/blob/582e404a1041ce95d22939b73f0b4d95be77c6ec/lib/lib.es2020.promise.d.ts#L21-L31
+export type PromiseSettledResult<ResolveValueType> = {
+	status: 'fulfilled';
+	value: ResolveValueType;
+} | {
+	status: 'rejected';
+	reason: any;
```

Comment:
Let's make it `unknown`. I think they're planning to make errors use `unknown` instead of `any` in TS too eventually.